### PR TITLE
Fill in json serialize and parse

### PR DESCRIPTION
**Pull request by StachuDotNet**
Changelog:

```
Language and Standard Library
- implement `Json::serialize<'a>` and `Json::parse<'a>`
```

Note: I was experimenting with the idea of exposing a JsonParseError and JsonSerializationError type. Might just keep those internal for now, but I'm aware they're currently in a bad/mixed state.

### CODE REVIEW

**Comment by StachuDotNet** [comment]
This is feeling pretty good. All happy paths seem to be working, and most error cases behave well-enough.

- [ ] deal with merge conflicts
- [ ] `let parse` should return JsonParseError as the Error type, and the stdlib fn (for now) should do the toString-ing. Make sure we're not "escaping" from this anywhere.
- [ ] update `serialize` to support a `JsonSerializationError` as the error type, and similarly ToString it, if needed, _in_ the stdlib fn
- [ ] revisit test cases - I don't think it's worth being exhaustive for now, but there may be some low-hanging fruit (i.e. package-defined enums)
- [ ] ask gpt for more test cases

File: backend/tests/TestUtils/FSI.fs [hunk 1]
```diff
@@ -0,0 +1,30 @@
+/// <summary>
```
Comment (pbiggar) [inline backend/tests/TestUtils/FSI.fs:1]: Did you intend to resurrect this?

Comment (StachuDotNet) [reply backend/tests/TestUtils/FSI.fs:1]: nope, will re-remove

File: backend/src/Parser/Parser.fs [hunk 2]
```diff
@@ -42,7 +43,10 @@ module DType =
     // with type args
     | "List", [ arg ] -> PT.TList(fromSynType arg)
     | "Option", [ arg ] -> PT.TOption(fromSynType arg)
-
+    | "Result", [ okArg; errorArg ] ->
+      PT.TResult(fromSynType okArg, fromSynType errorArg)
+    | "Tuple", first :: second :: theRest ->
```
Comment (pbiggar) [inline backend/src/Parser/Parser.fs:48]: This surprises me - is the F# compiler using the word "Tuple"?

**Comment by pbiggar** [comment]
Would love to get this merged ASAP, even if it's not in the final shape. Maybe add some CLEANUPs for what's left to do?

File: backend/src/Parser/Parser.fs (hunk 2 above)
Comment (StachuDotNet) [reply backend/src/Parser/Parser.fs:48]: Hmm
The context here is handling `Json.serialize<Tuple<int, string>>`.
I guess I could have done `Json.serialize<int * string>`, as F# might "expect".

**Pull request merged**